website/content/configuration/_advanced_bgp_configuration.md

```diff
@@ -0,0 +1,293 @@
+
+For example, this is a (somewhat contrived) definition for a peer that
+will only be used by machines:
+
+- With hostname `hostA` or `hostB`, or
+- That have the `rack=frontend` label, but _not_ the label `network-speed=slow`:
+
+```yaml
+apiVersion: metallb.io/v1beta2
+kind: BGPPeer
+metadata:
+  name: example
+  namespace: metallb-system
+spec:
+  myASN: 64512
+  peerASN: 64512
+  peerAddress: 172.30.0.3
+  peerPort: 180
+  nodeSelectors:
+  - matchLabels:
+      rack: frontend
+    matchExpressions:
+    - key: network-speed
+      operator: NotIn
+      values: [slow]
+  - matchExpressions:
+    - key: kubernetes.io/hostname
+      operator: In
+      values: [hostA, hostB]
+```
+
+### Announcing the Service from a subset of nodes
+
+It is possible to limit the set of nodes that are advertised as next hops to reach
+the service IPs. This is achieved by using the node selector in the `BGPAdvertisement` CR.
+
+{{<mermaid align="center">}}
+graph TD
+    metallBA-->|connects|torA(Router)
+    metallBA-->|announces|torA(Router)
+    metallBB-->|connects|torA(Router)
+    metallBB-->|announces|torA(Router)
+    metallBC-->|connects|torA(Router)
+
+    subgraph NodeA
+        metallBA("MetalLB<br>Speaker")
+    end
+    subgraph NodeB
+        metallBB("MetalLB<br>Speaker")
+    end
+    subgraph NodeC
+        metallBC("MetalLB<br>Speaker")
+    end
+
+{{< /mermaid >}}
+
+In this example, the service is announced only from NodeA and NodeB.
+Note that this feature is orthogonal to the [BGP peer node selector](#limiting-peers-to-certain-nodes),
+as it's performed at service (`IPAddressPool`) level.
+
+MetalLB will still follow the `BGPPeer` configuration
+to choose if a certain node must be peered with a certain router.
+
+In order to limit the set of nodes for a given advertisement, the node selector must be set:
+
+```yaml
+apiVersion: metallb.io/v1beta1
+kind: BGPAdvertisement
+metadata:
+  name: example
+spec:
+  ipAddressPools:
+  - first-pool
+  nodeSelectors:
+  - matchLabels:
+      kubernetes.io/hostname: NodeA
+  - matchLabels:
+      kubernetes.io/hostname: NodeB
+```
+
+In this way, all the IPs coming from `first-pool` will be reacheable only via `NodeA`
+and `NodeB`.
+
+### Configuring the BGP source address
+
+When a host has multiple network interfaces or multiple IP addresses
+configured on one interface, the host's TCP/IP stack usually selects
+the IP address that is used as the source IP address for outbound
+connections automatically. This is true also for BGP connections.
+
+Sometimes, the automatically-selected address may not be the desired
+one for some reason. In such cases, MetalLB supports explicitly
+specifying the source address to be used when establishing a BGP
+session:
+
+```yaml
+apiVersion: metallb.io/v1beta2
+kind: BGPPeer
+metadata:
+  name: example
+  namespace: metallb-system
+spec:
+  myASN: 64512
+  peerASN: 64512
+  peerAddress: 172.30.0.3
+  peerPort: 180
+  sourceAddress: 10.0.0.2
```

Comment:
If this is a IBGP connection do you want the peers on the same subnet? Also maybe use the default port number 179.

Reply:
Fixed